odoo/modules/db.py

```diff
@@ -65,16 +73,20 @@ def initialize(cr):
             info['license'],
             info['application'], info['icon'],
             info['sequence'], Json({'en_US': info['summary']})))
-        id = cr.fetchone()[0]
-        cr.execute('INSERT INTO ir_model_data \
-            (name,model,module, res_id, noupdate) VALUES (%s,%s,%s,%s,%s)', (
-                'module_'+i, 'ir.module.module', 'base', id, True))
+        row = cr.fetchone()
+        assert row
```

Comment:
`fetchone` returns a tuple or None (if there are no records). There is a lot of people forgetting about the None :)
Actually, it can be useful if you do: `while row := cr.fetchone()`.

Reply:
But for a known query `INSERT ... RETURING ...`
`fetchone` won't be `None` right? So we dont' have to check it.

And in other cases where the `fetchone` will be `None` I think we should skip some code or raise an error

I feel `assert` is more like detecting a coding error. And `assert` can raise error immediately in this function instead of causing problems in other places which is hard to debug.
for `fetchone` even if there is not assert, the error is immediately raised when we use the `row` in the same function.

Well I am OK to remain the `assert`, I just feel it is not so necessary to have it. I mean if it is treated as a best/recommended practice, we will get more code everywhere after `fetchone()` 🤔

Reply:
Logically yes, but typing does not know that (adding an assert makes sure the typing is aware as well).
I agree it is not "beautiful", however, we do not have that many `fetchone` in base and usually if you have an env, you would use `execute_query` which handles that already.

Reply:
My 2 cents (since I'm getting the notifications also :)):
Can we at least put a comment[^1] to make sure we make explicit why we have an unnecessary assert? TBH I'd be confused myself if I see such assert after an insert query :/

[^1]: `assert row  # for better typing hints`